src/ripple/protocol/impl/Feature.cpp

```diff
@@ -17,127 +17,285 @@
 */
 //==============================================================================
 
-#include <ripple/basics/contract.h>
 #include <ripple/protocol/Feature.h>
-#include <ripple/protocol/digest.h>
 
+#include <ripple/basics/Slice.h>
+#include <ripple/basics/contract.h>
+#include <ripple/protocol/digest.h>
+#include <boost/container_hash/hash.hpp>
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/key_extractors.hpp>
+#include <boost/multi_index/random_access_index.hpp>
+#include <boost/multi_index_container.hpp>
 #include <cstring>
 
 namespace ripple {
 
-//------------------------------------------------------------------------------
+enum class Supported : bool { no = false, yes };
```

Comment:
I'm a little concerned about declaring `Supported` at `namespace ripple` scope.  There are a huge number of things in the code base that are supported.  In this case we're only talking about whether an `Amendment` or `Feature` is supported.  Yeah, this declaration is only visible to humans in this file.  But the compiler leaks the name `ripple::Supported` out to the entire application.

Consider renaming this to `AmendmentSupported`, which I'll admit is a little unwieldy.  Another option would be to declare it as a member of the `Feature` class and use the class to limit the scope.  Or maybe you have a yet better idea.

I won't insist, I'm just a bit concerned.

Reply:
Wouldn't it only leak the name to parts of the application in the same TU?
I put it in the `detail` namespace. I can't say I'm entirely happy with the way the variable initialization looks now, but it works.

Reply:
I'm pretty sure the name leaks into the entire executable through the linker.

What you _could_ do is put the declaration in an unnamed namespace like this:
```
namespace
{
    enum class Supported : bool { no = false, yes };
}
```
According to cppreference, for an unnamed name space...
>  Its members have potential scope from their point of declaration to the end of the translation unit, and have internal linkage.

https://en.cppreference.com/w/cpp/language/namespace

I think an unnamed namespace would provide the smallest possible scope.

Reply:
Done.